velox/exec/PartitionedOutput.cpp

```diff
@@ -240,7 +245,7 @@ RowVectorPtr PartitionedOutput::getOutput() {
     for (auto& destination : destinations_) {
       bool atEnd = false;
       blockingReason_ = destination->advance(
-          kMaxDestinationSize,
+          maxBufferedBytes_ / destinations_.size(),
```

Comment:
I wonder if it is cleaner to specify the max-size in destination's constructor and move the logic of randomizing the limit to a single place which creates all the destinations.